installer.py

```diff
@@ -407,135 +393,192 @@ def installLogos10():
             pass
 
-        if logos_acknowledge_question(f"A launch script has been placed in {config.INSTALLDIR} for your use. The script's name is {config.FLPRODUCT}.sh.\nDo you want to run it now?\nNOTE: There may be an error on first execution. You can close the error dialog.", "The Script has finished. Exiting…"):
-            subprocess.run([os.path.join(config.INSTALLDIR, f"{config.FLPRODUCT}.sh")])
-        logging.info("The script has finished. Exiting…")
+        # Copy executable to config.APPDIR.
+        runmode = get_runmode()
+        if runmode == 'binary':
+            launcher_exe = Path(f"{config.INSTALLDIR}/LogosLinuxLauncher")
+            # FIXME: Confirm file copy and test desktop launcher.
+            if launcher_exe.is_file():
+                logging.debug(f"Removing existing launcher binary.")
+                launcher_exe.unlink()
+            logging.info(f"Creating launcher binary by copying this installer binary to {launcher_exe}.")
+            shutil.copy(sys.executable, launcher_exe)
+            create_shortcuts()
+
+            if config.DIALOG == 'tk':
+                subprocess.Popen(str(launcher_exe))
+            elif logos_acknowledge_question(f"An executable has been placed at {launcher_exe}.\nDo you want to run it now?\nNOTE: There may be an error on first execution. You can close the error dialog.", "The Script has finished. Exiting…"):
+                subprocess.Popen([str(launcher_exe)])
+        elif runmode == 'script':
+            if config.DIALOG == 'tk':
+                subprocess.Popen(sys.argv)
+            elif logos_acknowledge_question(f"Run {config.FLPRODUCT} now?", "The Script has finished. Exiting…"):
+                subprocess.Popen(sys.argv)
+        message = "The Script has finished. Exiting…"
+        cli_msg(message)
+        logging.info(message)
     else:
-        logos_error(f"Installation failed. {config.LOGOS_EXE} not found. Exiting…\nThe {config.FLPRODUCT} executable was not found. This means something went wrong while installing {config.FLPRODUCT}. Please contact the Logos on Linux community for help.", "")
+        logos_error(f"Installation failed. {config.LOGOS_EXE} not found. Exiting…\nThe {config.FLPRODUCT} executable was not found. This means something went wrong while installing {config.FLPRODUCT}. Please contact the Logos on Linux community for help.")
 
 def install():
-    # BEGIN PREPARATION
+    prepare_install()
+    finish_install()
+
+def prepare_install():
     chooseProduct()  # We ask user for his Faithlife product's name and set variables.
     chooseVersion()  # We ask user for his Faithlife product's version, set variables, and create project skeleton.
     getLogosReleaseVersion(getLogosReleases())
     logos_setup() # We set some basic variables for the install, including retrieving the product's latest release.
     chooseInstallMethod()  # We ask user for his desired install method.
```

Comment:
I think these two functions need to be flipped, right?

Reply:
Maybe the names are misleading and were ill-chosen, but the order is correct. Basically, `prepare_install()` sets all the installer choices, then `finish_install()` applies everything. So curses calls `install()`, which then calls `prepare_install()` and `finish_install()`, but the Tk window sets all the choices in the GUI window then calls `finish_install()` directly. It doesn't use `install()` or `prepare_install()` at all.

Reply:
Ah, in this case, I meant are they defined in proper order. I would have assumed prepare_install would need to be defined before install.